### PR TITLE
[data] [streaming] No preserve order by default

### PR DESCRIPTION
**Pull request by ericl**
Signed-off-by: Eric Liang <ekhliang@gmail.com>

## Why are these changes needed?

Preserve order decreases performance; set it off by default.

### CODE REVIEW

File: python/ray/data/_internal/execution/interfaces.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ class ExecutionOptions:
 
     # Always preserve ordering of blocks, even if using operators that
```
Comment (jianoaix) [inline python/ray/data/_internal/execution/interfaces.py:153]: nit: the comment doesn't match now.

**Review by jianoaix — COMMENTED** [review]
LG for performance, as long as the most use case doesn't actually need order preservation.

**Review by clarkzinzow — APPROVED** [review]
LGTM! I'm guessing that we'll eventually expose this as an AIR-level config, and maybe make the default for streaming training ingest to be `True`?

**Comment by ericl** [comment]
I think we'd still default to false, but yeah it should be exposed in the AIR config.

**Pull request merged**